NotificationHubs/src/com/windowsazure/messaging/RetryPolicy.java

```diff
@@ -10,16 +10,18 @@
 import java.util.concurrent.TimeoutException;
 
 /**
- * An abstract representation of a policy to govern retrying of messaging operations.
+ * Policy to govern retrying of operations.
  */
-public abstract class RetryPolicy {
+public class RetryPolicy {
```

Comment:
I'm a bit unclear on why we deleted the existing concrete classes and took the `abstract` off of this one. Could you explain a bit more?

Reply:
I can't explain why. These changes was added before me.

Reply:
Before "align with retry policy ADR" updates we used to create retry policy of a certain type (fixed or exponential) on hub or namespace manager initialization, based on options provided.
But according to ADR retry policy pattern we should be able to change dynamically retry policy mode in dependence on type of the error, e.g. trottling error should have linear backoff, whereas another retryable nh error should apply user defined mode if present or should have exponential delay.

Reply:
Thanks for the explanation! After reading through the code a bit more, I actually liked the hierarchy they had pretty well. I'm going to push a commit or two here shortly to bring this PR a little closer to the original code. I appreciate this though.